power_grid_model_c/power_grid_model/include/power_grid_model/three_phase_tensor.hpp

```diff
@@ -9,11 +9,13 @@
 #include "power_grid_model.hpp"
 
 // eigen properties
+// clang-format off
 #include <Eigen/Dense>
```

Comment:
I guess `Eigen` should have it's own block (separated by whitelines) together with other external libraries `boost`, `nlohmann` and `doctest`. that may be one potential argument in favour of the `IncludeCategories` (but as I said before, I prefer not to add that, so if it autodetects, that would be great)